website/source/docs/guides/agent-encryption.html.md

```diff
@@ -0,0 +1,205 @@
+  "server": true,
+  "encrypt": "JY34uTPZyfUE+6tinMYEVw==",
+  "encrypt_verify_incoming": false,
+  "encrypt_verify_outgoing": false
+}
+```
+
+A rolling update can be made by reloading the Consul process, `consul reload` or `kill -HUP <process_id>`.
+
+**Step 3**: Update the `encrypt_verify_outgoing` setting to `true` and perform another rolling update of the cluster. The agents will now be sending encrypted gossip but will still allow incoming unencrypted traffic.
+
+```javascript
+{
+  "data_dir": "/opt/consul",
+  "log_level": "INFO",
+  "node_name": "bulldog",
+  "server": true,
+  "encrypt": "JY34uTPZyfUE+6tinMYEVw==",
+  "encrypt_verify_incoming": false,
+  "encrypt_verify_outgoing": true
+}
+```
+
+Again, use `consul reload` or `kill -HUP <process_id>` to reload the process. 
+
+**Step 4**: The previous step, enabling verify outgoing, must be completed on all agents before continuing. Update the `encrypt_verify_incoming` setting to `true` and perform a final rolling update of the cluster. 
+
+```javascript
+{
+  "data_dir": "/opt/consul",
+  "log_level": "INFO",
+  "node_name": "bulldog",
+  "server": true,
+  "encrypt": "JY34uTPZyfUE+6tinMYEVw==",
+  "encrypt_verify_incoming": true,
+  "encrypt_verify_outgoing": true
+}
+```
+
+All the agents will now be strictly enforcing encrypted gossip.
+
+## TLS Encryption for RPC
+
+Consul supports using TLS to verify the authenticity of servers and clients. To enable TLS,
+Consul requires that all clients and servers have key pairs that are generated by a single
+Certificate Authority. The CA then signs keys for each of the agents, as in
+[this tutorial on generating both a CA and signing keys](/docs/guides/creating-certificates.html)
+using [cfssl](https://cfssl.org/).
+
+TLS can be used to verify the authenticity of the servers with [`verify_outgoing`](/docs/agent/options.html#verify_outgoing) and verify the authenticity of clients with [`verify_server_hostname`](/docs/agent/options.html#verify_server_hostname).
+
+Review the [docs for specifics](https://www.consul.io/docs/agent/encryption.html).
+
+In Consul version 0.8.4 and newer, migrating to TLS encrypted traffic on a running cluster
+is supported. 
+
+### Enable TLS: New Cluster
+
+After TLS has been configured on all the agents, you can start the agents and RPC communication will be encrypted.
+
+```javascript
+{
+  "data_dir": "/opt/consul",
+  "log_level": "INFO",
+  "node_name": "bulldog",
+  "server": true,
+  "encrypt": "JY34uTPZyfUE+6tinMYEVw==",
+  "verify_incoming": true,
+  "verify_outgoing": true,
+  "verify_server_hostname": true,
+  "ca_file":,
+  "cert_file":,
+  "key_file":
+}
+```
+
+The `verify_outgoing` parameter enables agents to verify the authenticity of Consul for outgoing connections. The `verify_server_hostname` parameter requires outgoing connections to perform hostname verification. Finally, the `verify_incoming` parameter enables the servers to verify the authenticity of all incoming connections. 
```

Comment:
```suggestion
The `verify_outgoing` parameter enables agents to verify the authenticity of Consul servers for outgoing connections. The `verify_server_hostname` parameter requires outgoing connections to perform hostname verification and is critically important to prevent compromised client agents from becoming servers and revealing all state to the attacker. Finally, the `verify_incoming` parameter enables the servers to verify the authenticity of all incoming connections. 
```

Maybe also add another link here to Hans' new guide which covers all the other subtelties here like allowing UI access etc?